packages/grid/_modules_/grid/hooks/features/clipboard/useGridClipboard.ts

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import { ownerDocument } from '@material-ui/core/utils';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridApiEventHandler } from '../../root/useGridApiEventHandler';
+import { GRID_KEYDOWN } from '../../../constants/eventsConstants';
+import { buildCSV } from '../export/serializers/csvSerializer';
+import { useGridSelector } from '../core/useGridSelector';
+import { visibleGridColumnsSelector } from '../columns/gridColumnsSelector';
+import { gridCheckboxSelectionColDef } from '../../../models/colDef';
+import { GridClipboardApi } from '../../../models/api';
+import { useGridApiMethod } from '../../root/useGridApiMethod';
+
+export const useGridClipboard = (apiRef: GridApiRef): void => {
+  const visibleColumns = useGridSelector(apiRef, visibleGridColumnsSelector);
+
+  const writeToClipboardPolyfill = React.useCallback(
+    (data: string) => {
+      const doc = ownerDocument(apiRef.current.rootElementRef!.current!);
+      const span = doc.createElement('span');
+      span.style.whiteSpace = 'pre';
+      span.style.userSelect = 'all';
+      span.style.opacity = '0px';
+      span.textContent = data;
+
+      apiRef.current.rootElementRef!.current!.appendChild(span);
```

Comment:
is there a specific for appending in the data grid and not in the document directly? I'm asking so we could have this method be outside of the scope of the hook.

Reply:
I never considered appending to the document, but we might have a reason to not do it. If the document doesn't have a `body` tag we can't append. I know it's a pretty absurd case, but it may happen.

Reply:
It's safe, see https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/utils/getScrollbarSize.js, we never had any issue with it

Reply:
Updated.